components/ProposalCard.tsx

```diff
@@ -81,55 +87,65 @@ export const RenderProposalContent = ({
   } else if ("executeLambda" in content) {
     const metadata = JSON.parse(content.executeLambda.metadata ?? "{}");
 
-    if (
-      !metadata?.contract_address &&
-      !metadata?.meta?.includes("contract_addr") &&
-      !metadata?.meta?.includes("baker_address") &&
-      !metadata?.meta?.includes("fa1_2_address")
-    ) {
+    const parser = new Parser();
+
+    const [type, lambda] = parseLambda(
+      parser.parseMichelineExpression(content.executeLambda.content ?? "")
+    );
+
+    if (type === LambdaType.LAMBDA_EXECUTION) {
       data = {
         ...data,
         label: "Execute lambda",
         metadata:
           metadata.meta === "No meta supplied" ? undefined : metadata.meta,
         params: metadata.lambda,
       };
-    } else if (metadata?.meta?.includes("fa1_2_address")) {
-      const contractData = JSON.parse(metadata.meta).payload;
+    } else if (
+      type === LambdaType.FA1_2_APPROVE ||
```

Comment:
Imo it's not that painful to write it like so and avoid writing 2 times `data = { ...`

Reply:
I think writing two times data is ok. it can make code look clear and you don't need to have many branch to determine which case it's inside. but ok it's style issue, let's focus on others.